src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -1614,6 +1614,27 @@ public GHRepository fork() throws IOException {
         throw new IOException(this + " was forked but can't find the new repository");
     }
 
+    /**
+     * Sync this repository fork branch
+     *
+     * @param branch
+     *            the branch to sync
+     * @return The current repository
+     * @throws IOException
+     *             the io exception
+     */
+    public GHRepository sync(String branch) throws IOException {
+        if (isFork()) {
```

Comment:
If someone tries to sync a repo that is not a fork, shouldn't we throw an exception?  Would the REST API throw an exception if this call were made against a repo that is not a fork?

Reply:
Hi,

Yes I get 422

```
org.kohsuke.github.HttpException: {"message":"Validation Failed","documentation_url":"https://docs.github.com/rest/branches/branches#sync-a-fork-branch-with-the-upstream-repository","status":"422"}
```

What do you prefer ? Remove the check or let the exception to the user ?

Reply:
Done in 5083df701414897ed171202aba5ea262a589d5e3